stdlib/public/core/Builtin.swift

```diff
@@ -397,7 +397,13 @@ internal var _objectPointerIsObjCBit: UInt {
 #else
 @inlinable
 internal var _objectPointerIsObjCBit: UInt {
-  @inline(__always) get { return 0x4000_0000_0000_0000 }
+  @inline(__always) get {
+#if os(Android) && arch(arm64)
+    return 0x0040_0000_0000_0000
```

Comment:
I don't think this should be used because ObjC interoperability is disabled on Android, same for `largeIsCocoa()` and `largeCocoa()` below, but just played it safe by shifting these too, as some code may still call these somewhere.

Reply:
Is it possible to phrase that in terms of an assertion (i.e. `_invariantCheck`)? That way it'll at least show up when testing.

Reply:
I'll look into it.

Reply:
I checked and this function and `largeCocoa()` are unused when running the compiler validation suite but almost 200 stdlib tests do check `largeIsCocoa()`, so I added these `_internalInvariantFailure()` traps for these two functions that are unused on Android AArch64. That way, everything is shifted properly when assertions are turned off, but these two functions will fail when assertions are turned on in Android AArch64, to warn anyone invoking them not to use them.

Reply:
> Should we just have an unreachable statement instead?

I'm fine with anything for these two unused methods `_objectPointerIsObjCBit` and `largeCocoa()` at this point. I can leave them untouched, trap, or apply any other mixed approach like this that you want.

I had initially modified these two methods only because I didn't know if they were used somewhere, but now that a full run of the compiler validation suite on a debug+asserts build of the stdlib for Android AArch64 passed, I'm reasonably confident these two are completely unused and can revert all changes to them if you prefer.

Just let me know what you think is best.

Reply:
I will revert all my changes to these two ObjC-related methods.

Reply:
Done.